src/pages/blocks/[blockId]/index.tsx

```diff
@@ -0,0 +1,211 @@
+import { OverflowTable } from '@components/commons/OverflowTable'
+import { CursorPage, CursorPagination } from '@components/commons/CursorPagination'
+import { getWhaleApiClient } from '@contexts/WhaleContext'
+import { IoCopyOutline } from 'react-icons/io5'
+
+import { fromUnixTime, format } from 'date-fns'
+
+interface BlockDetailsPageProps {
+  block: Block
+  confirmations: number
+  transactions: {
+    items: Transaction[]
+    pages: CursorPage[]
+  }
+}
+
+function CopyButton ({ text }: { text: string }): JSX.Element {
+  const [open, setOpen] = useState<Boolean>(false)
+
+  async function copy (): Promise<void> {
+    await navigator.clipboard.writeText(text)
+    setOpen(true)
+
+    setTimeout(() => {
+      setOpen(false)
+    }, 500)
+  }
+
+  return (
+    <div className='ml-1'>
+      <button className='cursor-pointer outline-none p-1 bg-gray-100 rounded shadow-sm' onClick={copy}>
+        <IoCopyOutline className='h-5 w-5 text-gray-500' />
+      </button>
+      {
+        open === true
+          ? (
+            <span className='bg-gray-100 p-1 rounded absolute mr-4'>
+              Copied!
+            </span>
+          )
+          : null
+      }
+    </div>
+  )
+}
+
+export default function BlockDetails ({ block, confirmations, transactions }: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  const leftBlockDetails = [
+    // {
+    //   label: 'Block Reward:',
+    //   value: 'xxxx DFI'
+    // },
+    {
+      label: 'Height:',
+      value: `${block.height}`,
+      testId: 'block-detail-height'
+
+    },
+    {
+      label: 'Transactions:',
+      value: `${block.transactionCount}`,
+      testId: 'block-detail-transactions'
+    },
+    {
+      label: 'Timestamp:',
+      value: `${format(fromUnixTime(block.medianTime), 'PPpp')}`,
+      testId: 'block-detail-timestamp'
+    },
+    {
+      label: 'Confirmations:',
+      value: `${confirmations}`,
+      testId: 'block-detail-confirmations'
+    },
+    {
+      label: 'Merkle Root:',
+      content: (
+        <>
+          <span className='flex-grow break-all' data-testid='block-detail-merkle-root'>{block.merkleroot}</span>
+          <CopyButton text={block.merkleroot} />
+        </>
+      )
+    }
+  ]
+
+  const rightBlockDetails = [
+    {
+      label: 'Difficulty:', value: `${block.difficulty}`, testId: 'block-detail-difficulty'
+    },
+    // {
+    //   label: 'Bits:', value: `${block.weight}`
+    // },
+    {
+      label: 'Size (bytes):', value: `${block.size}`, testId: 'block-detail-size'
+    },
+    {
+      label: 'Version:', value: `${block.version}`, testId: 'block-detail-version'
+    }
+    // {
+    //   label: 'Next Block:', value: `${block.nextBlock.height}`
+    // },
+    // {
+    //   label: 'Previous Block:', value: `${block.prevBlock.height}`
+    // }
+
+  ]
+
+  function renderBlockDetails (details: Array<{ label: string, content?: JSX.Element, value?: string, testId?: string }>): JSX.Element[] {
```

Comment:
Don't pass Array into a component and create a list of Arrayed Element[]

Loop the Array instead and map the Component.